### DISCUSSION TITLE
Where can I find the Keycloak legacy (version 18) documentation ?

### DISCUSSION THREAD

**Discussion by omasseau** [keycloak/keycloak#23200] [Q&A]
Hello,

I noticed that Keycloak documentation for legacy version (in my case Keycloak 18) is not available anymore on keycloak.org  : 
https://www.keycloak.org/documentation-archive.html

The problem is that I need this documentation.
Our clients have legacy versions in production and we cannot just update like that their Keycloak to version 21 (this requires planning and testing that can take several months).

I understand you do not support the legacy versions, but why removing the documentation ?

**Comment by justin-tay** [comment]
You can try https://web.archive.org/web/20221006034603/https://www.keycloak.org/docs/18.0/server_admin/

**Comment by omasseau** [comment]
Thanks. Still I don't understand this decision to remove the documentation of all the versions below 21.
Not supporting previous versions should not mean removing their documentation.
At least it would have been nice to warn users some months in advance through the blog.
We have in our own documentation many links to the Keycloak documentation. Now all these links are dead :(

**Comment by madrover** [comment]
It would be great to have some clarity regarding how long documentation will be available so user can plan accordingly.

I have noticed that since release `21.1.0` there's a documentation zip included with the release, is this the documentation archive mechanism going forward? See https://github.com/keycloak/keycloak/releases/download/21.1.0/keycloak-documentation-21.1.0.zip

**Reply by madrover** [reply]
Actually, I think this is related to https://www.keycloak.org/2023/04/keycloak-2110-released.html where it is stated that the [Documentation repository](https://github.com/keycloak/keycloak-documentation) has been deprecated. As part of this migration it seems that the access to previous version has been lost as well.

**Comment by quentin34** [comment]
Is there any chance to have oldest documentation available again?

**Reply by omasseau** [reply]
I've created an issue for this. Feel free to vote 🙂
https://github.com/keycloak/keycloak/issues/23210

**Comment by manologg** [comment]
This was also an inconvenient for me and did this: https://manologg.github.io/keycloak-documentation-18.0.2/
